pkg/rest/auth.go

```diff
@@ -24,39 +24,71 @@ type ConfigOAuth struct {
 
 // Auth holds the auth strategy for all Seeders
 type AuthConfig struct {
-	AuthStrategy AuthType          `yaml:"type"`
-	Username     string            `yaml:"username"`
-	Password     string            `yaml:"password"`
-	OAuth        *ConfigOAuth      `yaml:"oauth,omitempty"`
-	HttpHeaders  map[string]string `yaml:"httpHeaders,omitempty"`
+	AuthStrategy AuthType     `yaml:"type"`
+	Username     string       `yaml:"username"`
+	Password     string       `yaml:"password"`
+	OAuth        *ConfigOAuth `yaml:"oauth,omitempty"`
 }
 
-// Auth holds the auth strategy for all Seeders
-type Auth struct {
-	config           AuthConfig
-	oAuthConfig      *clientcredentials.Config
-	basicAuthToToken string
+type AuthMap map[string]AuthConfig
+
+// type Auth map[string]struct {
+// 	config           AuthMap
+// 	oAuthConfig      *clientcredentials.Config
+// 	basicAuthToToken string
+// }
+
+// Auth holds the auth strategy for each seeder
+type auth struct {
+	authStrategy AuthType
+	oAuthConfig  *clientcredentials.Config
+	basicAuth    *basicAuth
+	basicToToken *basicToToken
 }
 
-func NewAuth(a AuthConfig) *Auth {
-	ac := &Auth{
-		config: a,
-	}
-	if a.AuthStrategy == OAuth && a.OAuth != nil {
-		c := &clientcredentials.Config{
-			ClientID:       a.Username,
-			ClientSecret:   a.Password,
-			TokenURL:       a.OAuth.ServerUrl,
-			Scopes:         a.OAuth.Scopes,
-			EndpointParams: a.OAuth.EndpointParams,
-			AuthStyle:      oauth2.AuthStyleInParams,
-		}
-		if a.OAuth.OAuthSendParamsInHeader {
-			c.AuthStyle = oauth2.AuthStyleInHeader
+type basicAuth struct {
+	username string
+	password string
+}
+
+type basicToToken struct {
+	username     string
+	password     string
+	headerKey    string
+	headerValFmt string
+}
+
+type authMap map[string]auth
```

Comment:
Slightly confusing having two different types `AuthMap` and `authMap`

Reply:
name changed...